debezium-core/src/main/java/io/debezium/data/geometry/Point.java

```diff
@@ -6,72 +6,125 @@
 
 /**
  * A semantic type for a geometric Point, defined as a set of (x,y) coordinates.
- * 
+ *
+ * This historically used to be a useful class, but is now a MySQL-only type to be deprecated.
+ *
  * @author Horia Chiorean
  * @author Omar Al-Safi
  */
-public class Point {
+public class Point extends Geometry {
+
+    protected static final Logger LOGGER = LoggerFactory.getLogger(Point.class);
 
     public static final String LOGICAL_NAME = "io.debezium.data.geometry.Point";
     public static final String X_FIELD = "x";
     public static final String Y_FIELD = "y";
-    public static final String WKB_FIELD = "wkb"; //Please see DBZ-208
 
     /**
-     * Returns a {@link SchemaBuilder} for a Uuid field. You can use the resulting SchemaBuilder
-     * to set additional schema settings such as required/optional, default value, and documentation.
-     * 
+     * Returns a {@link SchemaBuilder} for a Point field.
+     * A Geometry with extra X & Y fields
+     *
      * @return the schema builder
      */
     public static SchemaBuilder builder() {
         return SchemaBuilder.struct()
-                            .name(LOGICAL_NAME)
-                            .version(1)
-                            .doc("A geometric point")
-                            .field(X_FIELD, Schema.FLOAT64_SCHEMA)
-                            .field(Y_FIELD, Schema.FLOAT64_SCHEMA)
-                            .field(WKB_FIELD, Schema.OPTIONAL_BYTES_SCHEMA);
+                .name(LOGICAL_NAME)
+                .version(1)
+                .doc("A Geometry (POINT)")
+                .field(WKB_FIELD, Schema.BYTES_SCHEMA)
+                .field(SRID_FIELD, Schema.INT32_SCHEMA)
+                .field(X_FIELD, Schema.FLOAT64_SCHEMA)
+                .field(Y_FIELD, Schema.FLOAT64_SCHEMA);
+    }
+
+    private static final int WKB_POINT = 1;  // type constant
+    private static final int WKB_POINT_SIZE = (1 + 4 + 8 + 8);  // fixed size
+
+    /**
+     * Creates WKB for a 2D {x,y} point.
+     * @param x coordinate
+     * @param y coordinate
+     * @return OGC WKB byte array
+     */
+    protected static byte[] buildWKBPoint(double x, double y) {
+        ByteBuffer wkb = ByteBuffer.allocate(WKB_POINT_SIZE);
+        wkb.put((byte)1); // BOM
+        wkb.order(ByteOrder.LITTLE_ENDIAN);
+
+        wkb.putInt(WKB_POINT);
+        wkb.putDouble(x);
+        wkb.putDouble(y);
+        return wkb.array();
     }
 
     /**
-     * Returns a {@link SchemaBuilder} for a Point, with all other default Schema settings.
-     * 
-     * @return the schema
-     * @see #builder()
+     * Parses a 2D WKB Point into a {x,y} coordinate array.
+     * Returns null for any non-point or points with Z/M/etc modifiers.
+     * @param wkb OGC WKB geometry
+     * @return x,y coordinate array
      */
-    public static Schema schema() {
-        return builder().build();
+    protected static double[] parseWKBPoint(byte[] wkb) throws IllegalArgumentException {
```

Comment:
@rcoup I am just curious, is there any reason why you didn't use the parser in [```MySqlGeometry.java```](https://github.com/debezium/debezium/blob/master/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlGeometry.java)?

Reply:
Felt unnecessary to pull in a full WKB parser lib for the simplest possible WKB of a 2D Point. If Debezium did need to understand the full type hierarchy/etc IMO we'd need JTS or GeoTools

Reply:
Thanks for clarifying this. I am personally not fan of having external libraries for simple tasks 👍  .